pkg/grpc/marshaler.go

```diff
@@ -0,0 +1,67 @@
+package grpc
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/protoadapt"
+)
+
+// Deprecated: customMarshaler is a hack for keeping backward compatibility for duration objects.
+// TODO(ROX-25678): This should be deleted in 4.8.
+type customMarshaler struct {
+	*runtime.JSONPb
+}
+
+func (c customMarshaler) Unmarshal(data []byte, v interface{}) error {
+	unmarshalError := protojson.Unmarshal(data, v.(proto.Message))
+	if unmarshalError == nil {
+		return nil
+	}
+	suppressCVERequest, ok := v.(*v1.SuppressCVERequest)
```

Comment:
We will have to do a similar explicit conversion for every proto message that contains the duration type. This doesn't seem scalable long term.

Reply:
Correct, that's the reason why I think it should be deprecated and remove ASAP cc @vjwilson